app/src/main/java/org/grammatek/simacorrect/spellcheckerservice/SimaCorrectSpellCheckerService.kt

```diff
@@ -77,21 +75,29 @@ class SimaCorrectSpellCheckerService : SpellCheckerService() {
             val retval = arrayOfNulls<SentenceSuggestionsInfo>(textInfos.size)
             for (i in textInfos.indices) {
                 val suggestionList: Array<SuggestionsInfo>
-                val suggestionsIndexes: Array<YfirlesturAnnotation.SuggestionIndexes>
-
+                val suggestionsIndices: Array<YfirlesturAnnotation.AnnotationIndices>
                 try {
-                    val response = ConnectionManager.correctSentence(textInfos[i].text)
-                    val ylAnnotation = YfirlesturAnnotation(response)
-                    suggestionList = ylAnnotation.getSuggestionsForAnnotatedWords(suggestionsLimit).toTypedArray()
-                    suggestionsIndexes = ylAnnotation.suggestionsIndexes.toTypedArray()
+                    var text = textInfos[i].text!!
+                    if (text.isBlank()) {
+                        continue
+                    }
```

Comment:
Good point, I'm uppercasing the first word so it can be properly spell checked. If it's in lowercase it will always get the "should be uppercase" annotation from Yfirlestur but that's not helpful, we want Yfirlestur to spell/grammar check the word for other errors. I take care of lowercasing the word elsewhere so there's no mismatch.

This code might be useless as of now since of the JSON added support where we can simply ignore capitalization. 

This will be addressed in the newest PR where the JSON support has been integrated.